crates/goose-cli/src/session/mod.rs

```diff
@@ -104,6 +108,40 @@ impl Session {
         Ok(())
     }
 
+    pub async fn list_prompts(&mut self) -> HashMap<String, Vec<String>> {
+        let prompts = self.agent.list_extension_prompts().await;
+        prompts
+            .into_iter()
+            .map(|(extension, prompt_list)| {
+                let names = prompt_list.into_iter().map(|p| p.name).collect();
+                (extension, names)
+            })
+            .collect()
+    }
+
+    pub async fn get_prompt_info(&mut self, name: &str) -> Result<Option<output::PromptInfo>> {
+        let prompts = self.agent.list_extension_prompts().await;
+
+        // Find which extension has this prompt
+        for (extension, prompt_list) in prompts {
+            if let Some(prompt) = prompt_list.iter().find(|p| p.name == name) {
+                return Ok(Some(output::PromptInfo {
+                    name: prompt.name.clone(),
+                    description: prompt.description.clone(),
+                    arguments: prompt.arguments.clone(),
+                    extension: Some(extension),
+                }));
+            }
+        }
+
+        Ok(None)
+    }
+
+    pub async fn get_prompt(&mut self, name: &str, arguments: Value) -> Result<Vec<PromptMessage>> {
```

Comment:
if we invoked a PromptMessage that starts with assistant role - i think that would error because the CLI assumes its user msg and we also require alternative between user & assistant. can you check this? do we wanna enforce some checks here?

Reply:
added enforcement for user first, and that the messages alternate user & assistant